overlord/snapstate/snapmgr.go

```diff
@@ -560,6 +604,12 @@ func (m *SnapManager) Ensure() error {
 
 	m.runner.Ensure()
 
+	// needs to run after the runner
```

Comment:
This should be fine (unless I miss something). The generated apparmor profile is loaded during early boot automatically via /etc/init.d/apparmor. This ensures its available early on and its not more costly than any other of the profiles in that dir.